File: String/1678. Goal Parser Interpretation/solution.py
class Solution:
    def interpret(self, command: str) -> str:
                
        answer = ''

        i = 0
        
        while i < len(command):
            if command[i] == 'G':
                answer += 'G'
                i+=1
            elif command[i] == '(' and command[i+1] == ')':
                answer += 'o'
                i+=2 
            elif command[i] == '(' and command[i:i+4] != ')':
                answer += 'al'
                i+=4 
                
        return answer

if __name__ == '__main__':

    """ Example 1: """
    command = "G()(al)"
    print(Solution().interpret(command=command))

    """ Example 2: """
    command = "G()()()()(al)"
    print(Solution().interpret(command=command))

    """ Example 3: """
    command = "(al)G(al)()()G"
    print(Solution().interpret(command=command))

File: String/1678. Goal Parser Interpretation/solution.swift
class Solution {
    func interpret(_ command: String) -> String {
        
        var ans : String = ""
        
        var i : Int = 0
        
        while i < command.count {
                        
            if i+4 <= command.count {
                
                let start = command.index(command.startIndex, offsetBy: i)
                let end = command.index(command.startIndex, offsetBy: i+4)
                
                if command[start..<end] == "(al)" {
                    ans += "al"
                    i+=4
                    continue
                }
            }
            
            if i + 1 < command.count {
                
                if command[command.index(command.startIndex, offsetBy: i)] == "(", command[command.index(command.startIndex, offsetBy: i+1)] == ")" {
                    
                    ans += "o"
                    i+=2
                    continue
                }
                
           }
            
            if command[command.index(command.startIndex, offsetBy: i)] == "G" {
                ans += "G"
                i+=1
            }
            
        }
        
        return ans
    }
}

// Example 1:
let command = "G()(al)"
print(Solution().interpret(command))

// Example 2:
let command = "G()()()()(al)"
print(Solution().interpret(command))


// Example 3:
let command = "(al)G(al)()()G"
print(Solution().interpret(command))
